src/tools/wasm-interp.c

```diff
@@ -441,86 +401,148 @@ static void run_all_exports(WasmAllocator* allocator,
 
     if (WASM_SUCCEEDED(result)) {
-      if (s_verbose) {
-        wasm_disassemble_module(out_module, s_stdout_stream, 0,
-                                out_module->istream.size);
-      }
+      if (s_verbose)
+        wasm_disassemble_module(env, s_stdout_stream, *out_module);
 
-      result = wasm_init_interpreter_thread(allocator, out_module, out_thread,
-                                            &s_thread_options);
-      out_thread->host_func.callback = default_host_callback;
-      out_thread->host_func.user_data = NULL;
+      result = wasm_init_interpreter_thread(allocator, env, *out_module,
+                                            out_thread, &s_thread_options);
     }
     wasm_free(allocator, data);
   }
   return result;
 }
 
-static void destroy_module_and_thread(WasmAllocator* allocator,
-                                      WasmInterpreterModule* module,
-                                      WasmInterpreterThread* thread) {
-  wasm_destroy_interpreter_thread(allocator, thread);
-  wasm_destroy_interpreter_module(allocator, module);
+static void print_typed_values(WasmInterpreterTypedValue* values,
+                               size_t num_values) {
+  uint32_t i;
+  for (i = 0; i < num_values; ++i) {
+    print_typed_value(&values[i]);
+    if (i != num_values - 1)
+      printf(", ");
+  }
+}
+
+static WasmResult default_host_callback(const WasmInterpreterFunc* func,
+                                        const WasmInterpreterFuncSignature* sig,
+                                        uint32_t num_args,
+                                        WasmInterpreterTypedValue* args,
+                                        uint32_t num_results,
+                                        WasmInterpreterTypedValue* out_results,
+                                        void* user_data) {
+  memset(out_results, 0, sizeof(WasmInterpreterTypedValue) * num_results);
+  uint32_t i;
+  for (i = 0; i < num_results; ++i)
+    out_results[i].type = sig->result_types.data[i];
+
+  printf("called host " PRIstringslice "." PRIstringslice "(",
+         WASM_PRINTF_STRING_SLICE_ARG(func->host.module_name),
+         WASM_PRINTF_STRING_SLICE_ARG(func->host.field_name));
+  print_typed_values(args, num_args);
+  printf(") => (");
+  print_typed_values(out_results, num_results);
+  printf(")\n");
+  return WASM_OK;
+}
+
+static WasmResult spectest_import_func(WasmInterpreterImport* import,
+                                       WasmInterpreterFunc* func,
+                                       WasmInterpreterFuncSignature* sig,
+                                       void* user_data) {
+  func->host.callback = default_host_callback;
+  return WASM_OK;
+}
+
+static WasmResult spectest_import_table(WasmInterpreterImport* import,
+                                        WasmInterpreterTable* table,
+                                        void* user_data) {
+  return WASM_ERROR;
```

Comment:
I assume that the next few functions are not yet implemented (and hence return WASM_ERROR)?